src/System.Net.WebSockets.Client/tests/CancelTest.cs

```diff
@@ -125,19 +124,12 @@ public class CancelTest : ClientWebSocketTestBase
         {
             using (ClientWebSocket cws = await WebSocketHelper.GetConnectedWebSocket(server, TimeOutMilliseconds, _output))
             {
-                var cts = new CancellationTokenSource(500);
-
                 var recvBuffer = new byte[100];
                 var segment = new ArraySegment<byte>(recvBuffer);
 
-                try
-                {
-                    await cws.ReceiveAsync(segment, cts.Token);
-                    Assert.True(false, "Receive should not complete.");
-                }
-                catch (OperationCanceledException) { }
-                catch (ObjectDisposedException) { }
-                catch (WebSocketException) { }
+                var cts = new CancellationTokenSource(100);
+                // If this fails with OperationCancelledException, it was not yet being awaited - rerun test.
```

Comment:
Then I need to rephrase the comment.
Going by the method name (and internals), what the method is _attempting_ to test is that cancellation that occurs once `await`ing has started throws a `WebSocketException`.  It's trying for a completely different code path.  So if we get an OCE, we didn't hit the target path, and the test is inconclusive (hence, the re-run instruction).  100 ms is an eternity to a machine - I get `WebSocketException` even if I set it to **1** ms.
Perhaps something like:
> If cancellation occurs after ReceiveAsync returns the awaitable task, it should throw WebSocketException, and not OperationCanceledException.  If this assert fails, it's possibly because of simple timing issues; try re-running the test.